dev/NavigationView/NavigationView.cpp

```diff
@@ -1339,103 +1568,52 @@ void NavigationView::RaiseSelectionChangedEvent(winrt::IInspectable const& nextI
 // If nextItem is selectionsuppressed, we should undo the selection. We didn't undo it OnSelectionChange because we want change by API has the same undo logic.
 void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const winrt::IInspectable& nextItem)
 {
-    auto nextActualItem = nextItem;
-    if (!m_shouldIgnoreNextSelectionChange)
-    {
-        auto scopeGuard = gsl::finally([this]()
-        {
-            m_shouldIgnoreNextSelectionChange = false;
-        });
-        m_shouldIgnoreNextSelectionChange = true;
+    bool isSettingsItem = IsSettingsItem(nextItem);
+    bool isSelectionSuppressed = IsSelectionSuppressed(nextItem);
```

Comment:
>isSelectionSuppressed [](start = 9, length = 21)

looks like this is only used in the if statement below, just call IsSelectionSuppressed there